### ISSUE TITLE
Evidential deep learning and other feature requests

### ISSUE THREAD

**Issue by schinto** [microsoft/Graphormer#23]
Dear Graphormer authors,

thanks for this great piece of software!
I have some feature requests.

Can you please add the functionality for evidential deep learning?
See article:
[ACS Cent. Sci. 2021, 7, 8, 1356–1367](https://doi.org/10.1021/acscentsci.1c00546) 

Please add the 10 smaller datasets from MoleculeNet to the benchmarks. They are ogbg-moltox21, ogbg-molbace, ogbg-molbbbp, ogbg-molclintox, ogbg-molmuv, ogbg-molsider, and ogbg-moltoxcast for (multi-task) binary classification, and ogbg-molesol, ogbg-molfreesolv, and ogbg-mollipo for regression.
See [https://ogb.stanford.edu/docs/graphprop/](https://ogb.stanford.edu/docs/graphprop/)

Please add functionality for molecular representation pre-training via attribute masking
See [Strategies for Pre-training Graph Neural Networks](https://arxiv.org/pdf/1905.12265.pdf)

Please add metrics described in the [Regression Metrics Guide](https://www.h2o.ai/blog/regression-metrics-guide/)

As the manual selection of parameters for a graph neural network is difficult, please add support
for some of the automated machine learning techniques.
See for example techniques described in [AutoGL](https://github.com/THUMNLab/AutoGL)

Many thanks.
